src/Nethermind/Nethermind.Network.Discovery/DiscoveryApp.cs

```diff
@@ -100,15 +101,16 @@ public void Start()
 
     public async Task StopAsync()
     {
+        if (_logger.IsDebug) _logger.Debug("Stopping discovery timer");
+        if (_logger.IsDebug) _logger.Debug("Stopping discovery persistence timer");
+
         _appShutdownSource.Cancel();
-        StopDiscoveryTimer();
-        StopDiscoveryPersistenceTimer();
 
         if (_storageCommitTask is not null)
         {
             await _storageCommitTask.ContinueWith(x =>
             {
-                if (x.IsFaulted)
+                if (x.IsFaulted && x.Exception is not null && x.Exception.InnerException is not TaskCanceledException && x.Exception.InnerExceptions.All(ex => ex is not TaskCanceledException))
```

Comment:
There is similar condition in `BlockDownloader`, maybe let's do an extension method?